### PR TITLE
refactor(ui): Adding caching for users, groups, and roles

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

Adding caching to users, groups, and roles tabs. 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/identity/user/cacheUtils.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                start: currData?.listUsers?.start || 0,
+                count: (currData?.listUsers?.count || 1) - 1,
+                total: (currData?.listUsers?.total || 1) - 1,
+                users: newUsers,
+            },
+        },
+    });
+};
+
+const DEFAULT_LIST_SIZE = 25;
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/identity/user/cacheUtils.ts:38]: Can we move this constant to the top of the file?

File: datahub-web-react/src/app/identity/user/cacheUtils.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+export const clearUserListCache = (client) => {
+    client.cache.evict({
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/identity/user/cacheUtils.ts:41]: Nice!

File: datahub-web-react/src/app/identity/user/UserList.tsx [hunk 3]
```diff
@@ -42,7 +43,6 @@ export const UserList = () => {
 
     const [page, setPage] = useState(1);
     const [isViewingInviteToken, setIsViewingInviteToken] = useState(false);
-    const [removedUrns, setRemovedUrns] = useState<string[]>([]);
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/identity/user/UserList.tsx:45]: Glad we're getting rid of this :)

**Review by aditya-radhakrishnan — APPROVED** [review]
LGTM! Suggested some small fixes

File: datahub-web-react/src/app/identity/user/cacheUtils.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { ListUsersDocument, ListUsersQuery } from '../../../graphql/user.generated';
+
+export const removeUserFromListUsesCache = (urn, client, page, pageSize, query) => {
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/identity/user/cacheUtils.ts:3]: thanks good catch!

File: datahub-web-react/src/app/permissions/roles/ManageRoles.tsx [hunk 5]
```diff
@@ -118,6 +119,7 @@ export const ManageRoles = () => {
                     });
                     setTimeout(function () {
                         rolesRefetch();
+                        clearUserListCache(client);
                     }, 3000);
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/permissions/roles/ManageRoles.tsx:123]: so actually i think it's fine... 2 seconds still is sometimes not enough i've seen locally :/ ideally we actually have caching here as well

**Pull request merged**